app/services/bet_recommendation_generator.rb

```
@@ -0,0 +1,229 @@
+  def expected_value(win_prob, odds)
+    return 0 unless odds && win_prob && odds != 0
+
+    payout = odds.positive? ? odds / 100.0 : 100.0 / odds.abs
+    ev = (win_prob * payout) - (1 - win_prob)
+    ev.round(4)
+  end
+
+  ##
+  # Creates or updates a bet recommendation record for the current game and bet type with the provided attributes.
+  # Marks existing recommendations of the same bet type as not current if the ratings configuration version is current.
+  # @param attrs [Hash] Attributes for the recommendation, including bet type, prediction and odds IDs, team, line, odds, model value, confidence, expected value, recommendation status, and ratings config version.
```

RuboCop (1.75.5)

[convention] Line is too long. [213/140]

(Layout/LineLength)